dgl_sparse/src/sparse_matrix.cc

```diff
@@ -122,6 +124,63 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::FromDiag(
   return SparseMatrix::FromDiagPointer(diag, value, shape);
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::IndexSelect(
+    int64_t dim, torch::Tensor ids) {
+  auto id_array = TorchTensorToDGLArray(ids);
+  if ((!dim && HasCSR()) || (dim && HasCSC()) || !HasCOO()) {
+    auto csr = dim == 0 ? this->CSRPtr() : this->CSCPtr();
+    auto slice_csr = dgl::aten::CSRSliceRows(CSRToOldDGLCSR(csr), id_array);
+    auto slice_value =
+        this->value().index_select(0, DGLArrayToTorchTensor(slice_csr.data));
+    slice_csr.data = dgl::aten::NullArray();
```

Comment:
Why do you explicitly reset the array here.

Reply:
Since when we use SparseMatrix::FromCOOPointer(coo, value, shape) to construct a COO matrix, the origin 'coo' can not contain a data array, otherwise it will occur an error. In the CSR format, it will not occur this error immediately, but if it call coo() function later, sparse will use the csr format to construct COO, input the csr data array, and occur this error. 
For example, if we use csr to constrcut a sparse and then want to see the row and col attributes of it, sparse will convert it to COO format and occur an error.

Reply:
Got it. Then it is better to convert it to the new sparse format and then remove the index array. Please also explain the reason for doing this in the code.